internal/policy/commit/check_gpg_identity.go

```diff
@@ -0,0 +1,160 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package commit
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/talos-systems/conform/internal/git"
+	"github.com/talos-systems/conform/internal/policy"
+)
+
+// GPGIdentityCheck ensures that the commit is cryptographically signed using known identity.
+type GPGIdentityCheck struct {
+	errors   []error
+	identity string
+}
+
+// Name returns the name of the check.
+func (g GPGIdentityCheck) Name() string {
+	return "GPG Identity"
+}
+
+// Message returns to check message.
+func (g GPGIdentityCheck) Message() string {
+	if len(g.errors) != 0 {
+		return g.errors[0].Error()
+	}
+
+	return fmt.Sprintf("Signed by %q", g.identity)
+}
+
+// Errors returns any violations of the check.
+func (g GPGIdentityCheck) Errors() []error {
+	return g.errors
+}
+
+// ValidateGPGIdentity checks the commit GPG signature for a known identity.
+func (c Commit) ValidateGPGIdentity(g *git.Git) policy.Check {
+	check := &GPGIdentityCheck{}
+
+	switch {
+	case c.GPG.Identity.GitHubOrganization != "":
+		members, err := getOrganizationMembers(context.Background(), c.GPG.Identity.GitHubOrganization)
+		if err != nil {
+			check.errors = append(check.errors, err)
+
+			return check
+		}
+
+		keyrings, err := getKeyring(context.Background(), members)
+		if err != nil {
+			check.errors = append(check.errors, err)
+
+			return check
+		}
+
+		entity, err := g.VerifyPGPSignature(keyrings)
+		if err != nil {
+			check.errors = append(check.errors, err)
+
+			return check
+		}
+
+		for identity := range entity.Identities {
+			check.identity = identity
+
+			break
+		}
+	default:
+		check.errors = append(check.errors, fmt.Errorf("no signature identity configuration found"))
+	}
+
+	return check
+}
+
+func getOrganizationMembers(ctx context.Context, organization string) ([]string, error) {
+	var members []struct {
+		Login string `json:"login"`
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("https://api.github.com/orgs/%s/members", organization), nil)
```

Comment:
We already use github.com/google/go-github module, so why not use it there?

Reply:
:shrug: probably it's easier? but I can look into that (if that is even available)

Reply:
got it updated